src/api.js

```diff
@@ -54,10 +56,110 @@ api.declare({
   // Publish message
   await this.publisher.purgeCache({provisionerId, workerType, cacheName});
 
+  try {
+    await this.CacheBuster.create({
+      workerType,
+      provisionerId,
+      cacheName,
+      before: new Date(),
+      expires: taskcluster.fromNow('1 day'),
+    });
+  } catch (err) {
+    if (err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+    let cb = await this.CacheBuster.load({
+      workerType,
+      provisionerId,
+      cacheName,
+    });
+
+    await cb.modify(cacheBuster => {
+      cacheBuster.before = new Date();
+      cacheBuster.expires = taskcluster.fromNow('1 day');
+    });
+  }
+
   // Return 204
   res.status(204).send();
 });
 
+api.declare({
+  method:   'get',
+  route:    '/purge-requests',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
+  name:     'allPurgeRequests',
+  output:   SCHEMA_PREFIX_CONST + 'purge-cache-request-list.json#',
+  title:    'All Open Purge Requests',
+  description: [
+    'This is useful mostly for administors to view',
+    'the set of open purge requests. It should not',
+    'be used by workers. They should use the purgeRequests',
+    'endpoint that is specific to their workerType and',
+    'provisionerId.',
+  ].join('\n'),
+}, async function(req, res) {
+  let continuation = req.query.continuationToken || null;
+  let limit = parseInt(req.query.limit || 1000, 10);
+  let openRequests = await this.CacheBuster.scan({}, {continuation, limit});
+  return res.reply({
+    cacheHit: false,
+    continuationToken: openRequests.continuation || '',
+    requests: _.map(openRequests.entries, entry => {
+      return {
+        provisionerId: entry.provisionerId,
+        workerType: entry.workerType,
+        cacheName: entry.cacheName,
+        before: entry.before.toJSON(),
+        expires: entry.expires.toJSON(),
+      };
+    }),
+  });
+});
+
+api.declare({
+  method:   'get',
+  route:    '/purge-requests/:provisionerId/:workerType',
+  name:     'purgeRequests',
+  output:   SCHEMA_PREFIX_CONST + 'purge-cache-request-list.json#',
```

Comment:
This should take a querystring parameter with `since=date-time` I'm sure we can find a regex matching `new Date().toJSON()`. Ideally, one that also allows for microseconds as python will generate.

This way we can return an empty list for most calls... by quickly filtering out requests that are really old.